### PR TITLE
ECOPROJECT-180 Extending the service worker to support disconnect req…

### PR DESCRIPTION
**Pull request by bardielle**
…uests

### CODE REVIEW

File: protocol/yggdrasil.proto [hunk 1]
```diff
@@ -16,6 +16,10 @@ service Dispatcher {
 service Worker {
     // Send is called by the dispatcher to send data to a worker.
     rpc Send (Data) returns (Receipt) {}
+
+    // Register is called by a worker to indicate it is ready and capable of
```
Comment (jakub-dzon) [inline protocol/yggdrasil.proto:20]: The comment doesn't describe `Disconnect` operation

File: protocol/yggdrasil.proto [hunk 2]
```diff
@@ -67,3 +71,9 @@ message Data {
 
 // A Receipt message is sent as a successful response to a Send method.
 message Receipt {}
+
+// A DisconnectionResponse message contains the result of a disconnection request.
+message DisconnectionResponse {
+    // Whether or not the worker accepted the disconnection request.
+    bool disconnected = 1;
```
Comment (jakub-dzon) [inline protocol/yggdrasil.proto:78]: In what scenario the worker may reject disconnection request? 
Both yggdrasil and workers mustn't refuse to disconnect from the control plane.

**Review by jakub-dzon — CHANGES_REQUESTED** [review]

File: protocol/yggdrasil.proto (hunk 2 above)
Comment (bardielle) [reply protocol/yggdrasil.proto:78]: So what should be the response?

Comment (jakub-dzon) [reply protocol/yggdrasil.proto:78]: Maybe empty, like `Receipt`?

**Review by jakub-dzon — APPROVED** [review]

**Pull request merged**